### PR TITLE
Wp 920 full name lookup

### PR DESCRIPTION
**Pull request by DrPyser**
https://wazo-dev.atlassian.net/browse/WP-920

### CODE REVIEW

File: integration_tests/suite/test_wazo_user_backend.py [hunk 1]
```diff
@@ -75,9 +75,11 @@ def backend_config(self):
         }
 
     def test_that_the_lookup_returns_the_expected_result(self):
-        results = self.backend.search('dyl')
+        search_terms = ['dyl', 'bob', 'bob ', 'bob dyl', ' dyl']
```
[the remark this answers is not in the document]
Comment (xxbillouxx) [reply integration_tests/suite/test_wazo_user_backend.py:78]: What's about users having : 
- a firstname like 'John Junior' and lastname like 'Doe'
- a firstname like 'John' and lastname like 'Wilson Brown'
- a firstname like 'John' and lastname like 'Wilson-Brown' ?

Should this be covered by this PR ? (if yes, can you add tests for these cases plz ?)

**Review by xxbillouxx — COMMENTED** [review]
I have some questions/comments :-)

File: integration_tests/suite/test_wazo_user_backend.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DrPyser) [reply integration_tests/suite/test_wazo_user_backend.py:78]: Given the logic used in this PR to perform the lookup is a simple string contains operation( `search_term in Contact[column_name]`), the expectations is that any substring of either `firstname`, `lastname` or `{firstname} {lastname}` would match successfully. I think these are represented by the test coverage(meaning that the logic under test is simple enough that we're basically testing python builtins, which we can assume are extensively tested already). 

The problem that is addressed by the PR is that full name search was simply not supported by dird, despite being supported by the underlying confd API. So you could lookup on first name , and separately on last name, but not on the combined `{firstname} {lastname}`. The issue was not a corner case relative to the comparison algorithm used for the lookup, simply some missing logic.

As the actual behavior also depends on the `confd` API itself supporting full name search, the corner cases would also depend on the confd API. Having more exhaustive test coverage here would not actually do much to catch real-world bugs since the more complex logic is in confd's database lookups.

**Review by xxbillouxx — APPROVED** [review]
LGTM, big thank you :)
